pennylane/gradients/pulse_gradient.py

```diff
@@ -0,0 +1,532 @@
+            \widetilde{C_j}((v_1, v_2), \tau)&=
+            \bra{\psi_{j}((v_1, v_2), \pi/4, \tau)} B
+            \ket{\psi_{j}((v_1, v_2), \pi/4, \tau)}\\
+            &-\bra{\psi_{j}((v_1, v_2), -\pi/4, \tau)} B
+            \ket{\psi_{j}((v_1, v_2), -\pi/4, \tau)} \\
+            \ket{\psi_{j}((v_1, v_2), x, \tau)}
+            &= U_{(v_1, v_2)}(T, \tau) e^{-i x H_j}U_{(v_1, v_2)}(\tau, 0)\ket{0}.
+
+        Here we used the partial derivatives
+
+        .. math::
+
+            \frac{\partial f_1}{\partial v_1}((v_1, v_2), t) &= 1\\
+            \frac{\partial f_1}{\partial v_1}((v_1, v_2), t) &= \cos(v_2 t) t \\
+            \frac{\partial f_1}{\partial v_2}((v_1, v_2), t) =
+            \frac{\partial f_2}{\partial v_1}((v_1, v_2), t) &= 0
+
+        and the fact that both :math:`H_1=Z\otimes Z` and :math:`H_2=I\otimes X`
+        have two unique eigenvalues and therefore admit a two-term parameter-shift rule
+        (see e.g. `Schuld et al. (2018) <https://arxiv.org/abs/1811.11184>`_).
+
+        As a second scenario, consider the single-qubit pulse generated by
+
+        .. math::
+
+            H((v_1, v_2), t) = v_1 \sin(v_2 t) X
+
+        together with the observable :math:`B=Z`.
+        You may already notice that this pulse can be rewritten as a :class:`~.RX` rotation,
+        because we have a single Hamiltonian term and the spectrum of :math:`H` consequently
+        will be constant up to rescaling.
+        In particular, the unitary time evolution under the Schrödinger equation is given by
+
+        .. math::
+
+            U_{(v_1, v_2)}(t_2, t_1) &=
+            \exp\left(-i\int_{t_1}^{t_2} \mathrm{d}\tau v_1 \sin(v_2 \tau) X\right)\\
+            &=\exp(-i\theta(v_1, v_2) X)\\
+            \theta(v_1, v_2) &= \int_{t_1}^{t_2} \mathrm{d}\tau v_1 \sin(v_2 \tau)\\
+            &=-\frac{v_1}{v_2}(\cos(v_2 t_2) - \cos(v_2 t_1)).
+
+        As the ``RX`` rotation satisfies a (non-stochastic) two-term parameter-shift rule,
+        we could compute the derivatives with respect to :math:`v_1` and :math:`v_2` by
+        implementing :math:`\exp(-i\theta(v_1, v_2) X)`, applying the two-term shift rule
+        and evaluating the classical Jacobian of the mapping :math:`\theta(v_1, v_2)`.
+
+        Using the stochastic parameter-shift rule instead will lead to approximation errors.
+        This is because the approximated integral not only includes the shifted circuit
+        evaluations, which do not depend on :math:`\tau` in this example, but also on the
+        classical Jacobian, which is *not* constant over :math:`\tau`.
+        Therefore, it is important to implement pulses in the simplest way possible.
+    """
+    # pylint:disable=unused-argument
+    if not qml.active_return():
+        raise NotImplementedError(
+            "The stochastic pulse parameter-shift gradient only supports the new "
+            "return type. Use qml.enable_return() to turn it on."
+        )
```

Comment:
I think it will not be worth the effort, actually. Because the new system will become default in 0.30, @rmoyard told me. And supporting the old system would require the changes made in #3697 to be duplicated for the old system, and that's a considerable amount of work. 
Correct me if I underestimate the inconvenience caused by having to switch to the new return type system :)

Reply:
If it is default by v0.30 then I agree there really is no need 💯